source/common/buffer/buffer_impl.h

```diff
@@ -41,6 +41,16 @@ class Slice {
     }
   }
 
+  // SliceData
+  bool isMutable() const override {
+    // Extracted slice data is immutable by default.
+    return false;
+  };
+  absl::Span<const uint8_t> getData() const override {
+    return {base_ + data_, reservable_ - data_};
+  };
+  absl::Span<uint8_t> getMutableData() override { return {nullptr, 0}; }
```

Comment:
Calls to this method should crash.

Reply:
Also, move this implementation to UnownedSlice.

Reply:
Crash as in ASSERT?

Reply:
I had it in UnownedSlice, but decided to move to Slice because:
1. It is the default pre-override behavior.
2. DummySlice in buffer_test.cc would have to change.

Reply:
Will use the following:
```
  absl::Span<uint8_t> getMutableData() override {
    RELEASE_ASSERT(isMutable(), "Not allowed to call getMutableData if slice is immutable");
    return {base_ + data_, reservable_ - data_};
  }
```

Reply:
Changed in commit adde208